halo2_proofs/src/plonk/prover.rs

```diff
@@ -353,16 +355,29 @@ where
                 );
 
                 // Add blinding factors to advice columns
-                for advice_values in &mut advice_values {
-                    for cell in &mut advice_values[unusable_rows_start..] {
-                        *cell = Scheme::Scalar::random(&mut rng);
+                for (column_index, advice_values) in column_indices.iter().zip(&mut advice_values) {
+                    if !witness.unblinded_advice.contains(column_index) {
+                        for cell in &mut advice_values[unusable_rows_start..] {
+                            *cell = Scheme::Scalar::random(&mut rng);
+                        }
+                    } else {
+                        #[cfg(feature = "sanity-checks")]
```

Comment:
Not a comment related to the PR itself. But we should really document what this feature does. As it is getting a broader scope (or it can)

cc: @han0110 @davidnevadoc @kilic